src/renderers/shaders/ShaderLib/meshlambert_frag.glsl.js

```diff
@@ -51,6 +56,18 @@ void main() {
 	// accumulation
 	reflectedLight.indirectDiffuse = getAmbientLightIrradiance( ambientLightColor );
 
+	#if defined(USE_SHADOWMAP) && NUM_HEMI_LIGHTS > 0
```

Comment:
Something like this:
```glsl
// accumulation
reflectedLight.indirectDiffuse = getAmbientLightIrradiance( ambientLightColor );

#include <lightmap_fragment>

#ifdef DOUBLE_SIDED

	reflectedLight.indirectDiffuse += ( gl_FrontFacing ) ? vIndirectFront : vIndirectBack;

	reflectedLight.directDiffuse = ( gl_FrontFacing ) ? vLightFront : vLightBack;

#else

	reflectedLight.indirectDiffuse += vIndirectFront;

	reflectedLight.directDiffuse = vLightFront;

#endif

reflectedLight.indirectDiffuse *= BRDF_Diffuse_Lambert( diffuseColor.rgb );

reflectedLight.directDiffuse *= BRDF_Diffuse_Lambert( diffuseColor.rgb ) * getShadowMask();
```

Reply:
I just want to clarify that the reason for the shadow and hemisphere light check are because its only when we use these features together that we need to move the final light color calculation to the fragment shader. Otherwise it's fine to precalculate the final light color entirely in the vertex shader. Let me know if you'd prefer the simplicity wins over the small performance save and I'll submit a simplified commit.

Reply:
I prefer to model the lighting correctly, and keep indirect separate from direct.

Reply:
Ok cool, I'll have the modified PR ready tomorrow morning. Thanks!